cmd/kubernetes/kubernetes_remove.go

```diff
@@ -52,53 +62,112 @@ var kubernetesRemoveCmd = &cobra.Command{
-
 			for _, v := range kuberneteList {
-				_, err = client.DeleteKubernetesCluster(v.ID)
+
+				// Delete the Kubernetes cluster
+				_, err := client.DeleteKubernetesCluster(v.ID)
 				if err != nil {
 					utility.Error("error deleting the kubernetes cluster: %s", err)
 					os.Exit(1)
 				}
+
+				/* Poll for the deletion status
+				This is required because, if we try to delete other things like firewalls before
+				the cluster is completely deleted, we will encounter errors like "database_firewall_inuse_by_cluster".
+				*/
+				for {
+					_, err := client.FindKubernetesCluster(v.Name)
+					if err != nil {
+						if errors.Is(err, civogo.ZeroMatchesError) {
+							break // Cluster is deleted
+						}
+					}
+				}
+
+				// Delete volumes if --delete-volumes flag is set
+				if deleteVolumes {
+					volumes, err := client.ListVolumesForCluster(v.ID)
+					if err != nil {
+						if !errors.Is(err, civogo.ZeroMatchesError) {
+							utility.Error("Error listing volumes for cluster %s: %s", v.Name, err)
+						}
+					} else {
+						for _, volume := range volumes {
+							_, err := client.DeleteVolume(volume.ID)
+							if err != nil {
+								utility.Error("Error deleting volume %s: %s", volume.Name, err)
+							}
+							fmt.Printf("%s volume deleted", volume.ID)
+						}
+					}
+				}
+
+				// Output volumes left behind if deleteVolumes flag is not set
+				if !deleteVolumes {
+					volumes, err := client.ListVolumesForCluster(v.ID)
+					if err != nil {
+						if !errors.Is(err, civogo.ZeroMatchesError) {
+							utility.Error("Error listing volumes for cluster %s: %s", v.Name, err)
+						}
+					} else if len(volumes) > 0 {
+						fmt.Fprintf(os.Stderr, "Volumes left behind for Kubernetes cluster %s:\n", v.Name)
+						for _, volume := range volumes {
+							fmt.Fprintf(os.Stderr, "- %s\n", volume.ID)
+						}
+						fmt.Fprintf(os.Stderr, "Consider using '--delete-volumes' flag next time to delete them automatically.\n")
+					}
+				}
+
+				// Delete firewalls if --keep-firewalls flag is not set
+				if !keepFirewalls {
+					firewall, err := client.FindFirewall(v.Name)
+					if err != nil {
+						if errors.Is(err, civogo.MultipleMatchesError) {
+							utility.Error("Error deleting the firewall: %v. Please delete the firewall manually by visiting: https://dashboard.civo.com/firewalls", err)
+						} else {
+							utility.Error("Error finding the firewall: %v", err)
+						}
+					} else if firewall.ClusterCount == 0 {
+						_, err := client.DeleteFirewall(firewall.ID)
+						if err != nil {
+							utility.Error("Error deleting firewall %s: %v", firewall.Name, err)
+						} else {
+							fmt.Fprintf(os.Stderr, "Firewall %s deleted. If you want to keep the firewall in the future, please use '--keep-firewalls' flag.\n", firewall.ID)
+						}
+					}
+				}
+
+				// Delete kubeconfig if --keep-kubeconfig flag is not set
+				if !keepKubeconfig {
+					kubeconfigPath := fmt.Sprintf("%s/.kube/config", os.Getenv("HOME"))
```

Comment:
you should not delete kubeconfig file because it may contain other clusters data even from different cloud providers.